.github/workflows/deploy-ghpages.yml

```diff
@@ -0,0 +1,63 @@
+name: Deploy Main to GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pages: write
+
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout main branch
+      uses: actions/checkout@v3
+
+    - name: Set up Rust
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+        override: true
+
+    - name: Install mdBook
+      run: cargo install mdbook
+
+    - name: Prepare src directory
+      run: |
+        mkdir -p src/vulnerabilities
+        cp README.md src/SUMMARY.md
+        cp -r vulnerabilities/* src/vulnerabilities/
+        
+    - name: Build the book
+      run: mdbook build
+
+    - name: Checkout gh-pages branch
+      uses: actions/checkout@v3
+      with:
+        ref: gh-pages
+        path: gh-pages
+
+    - name: Copy built book to gh-pages
+      run: |
+        rm -rf gh-pages/docs
+        mkdir -p gh-pages/docs
+        cp -r book/* gh-pages/docs/
+
+    - name: Commit and push changes
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I'm guessing I'll have to add an environment variable for this?

Reply:
> I'm guessing I'll have to add an environment variable for this?

Hi @kadenzipfel,

In the context of GitHub Actions, the GITHUB_TOKEN is automatically generated by GitHub for each workflow run. This token is unique to the repository and the specific workflow run, and it provides limited permissions to interact with the repository, such as pushing changes, creating issues etc. you can check the docs [here](https://docs.github.com/en/actions/security-guides/automatic-token-authentication)

You don't need to add an environment variable manually for this token. GitHub takes care of injecting this GITHUB_TOKEN into the workflow environment for you.